asm-to-pil/src/vm_to_constrained.rs

```diff
@@ -390,7 +392,7 @@ impl<T: FieldElement> VMConverter<T> {
                 param.index.is_none(),
                 "Cannot use array elements for instruction parameters."
             );
-            match &param.ty {
+            match param.ty.as_ref().and_then(|ty| ty.try_to_identifier()) {
```

Comment:
Same here, register should only be used if no type is given, not if it is not an identifier.